gcp/observability/config.go

```diff
@@ -34,73 +35,10 @@ const (
 	envObservabilityConfig     = "GRPC_CONFIG_OBSERVABILITY"
 	envObservabilityConfigJSON = "GRPC_CONFIG_OBSERVABILITY_JSON"
```

Comment:
The only env vars we have now are `GRPC_OBSERVABILITY_CONFIG` and `GRPC_OBSERVABILITY_CONFIG_FILE`.

Can we add these into envconfig instead?